pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -317,6 +343,7 @@ public ObjectNode toJsonObject() {
     }
     appendDefaultNullValue(jsonObject);
     appendTransformFunction(jsonObject);
+    jsonObject.put("notNull", !_nullable);
```

Comment:
this would be super error-prone going forward IMO

Reply:
Fixed in 802e9c4f93caa0815e74d85035847c1a2153f20f